### PR TITLE
Expand the use of ∈ and ∉ to SequenceType

### PR DESCRIPTION
**Pull request by LEOYoon-Tsaw**
with Equatable elements

### CODE REVIEW

File: stdlib/internal/SwiftExperimental/SwiftExperimental.swift [hunk 1]
```diff
@@ -118,13 +118,13 @@ public func ⨁= <
   lhs.exclusiveOrInPlace(rhs)
 }
 
-/// - Returns: True if `x` is in the set.
-public func ∈ <T>(x: T, rhs: Set<T>) -> Bool {
+/// - Returns: True if `x` is in the collection.
```
Comment (NachoSoto) [inline stdlib/internal/SwiftExperimental/SwiftExperimental.swift]: I imagine these should read "sequence" instead of "collection"?

Comment (LEOYoon-Tsaw) [inline stdlib/internal/SwiftExperimental/SwiftExperimental.swift]: Oh, that's true

**Comment by dduan** [comment]
Set (along with other containers shipped with Swift) is a value type with copy-on-write. So no copy gets made during these operations. That won't be necessarily true for arbitrary types implemented by user.

Further, expanding them on SequenceType is dangerous for its single-pass nature. For example, user could implement a SocketSequence, and ∈ would consume the entire socket. 

∈ and ∉ here refers to mathematical operators for set. IMHO, they should stay that way in Swift. `.contains` is suffice for `SequenceType`.

**Comment by LEOYoon-Tsaw** [comment]
If it's too ambitious to expand ∈ to SequenceType, expanding to IntervalType makes sense since then we can simply check interval like if x ∈ 0.0...0.9 rather than if x >= 0.0 && x <= 0.9 and improve code readability.

**Comment by dabrahams** [comment]
@dduan is right.
@LEOYoon-Tsaw you should actually write a proposal to have intervals and ranges of comparable indices conform to SetAlgebraType.  Then you’d get these operators for free.

**Pull request closed without merging**